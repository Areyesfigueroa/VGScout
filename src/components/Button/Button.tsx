import React from 'react';
import './Button.css';

const Button = (props) => (
    <div data-testid="button" className="button-style">
        {props.label}
    </div>
);

export default Button;