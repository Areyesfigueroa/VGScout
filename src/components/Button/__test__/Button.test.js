import React from 'react';
import ReactDOM from 'react-dom';

import Button from '../Button';

//React's UI testing library
import { render,cleanup } from '@testing-library/react';
import {toBeInTheDocument, toHaveStyle} from '@testing-library/jest-dom';

//Snapshots
import renderer from 'react-test-renderer';

expect.extend({toBeInTheDocument, toHaveStyle});

afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div);
}); 

it("renders click me button correctly", () => {
    const div = document.createElement("div");
    const {getByTestId} = render(<Button label="Click Me!"></Button>, div);
    expect(getByTestId('button')).toHaveTextContent("Click Me!");
});

it("renders save button correctly", () => {
    const div = document.createElement("div");
    const {getByTestId} = render(<Button label="Save"></Button>, div);
    expect(getByTestId('button')).toHaveTextContent("Save");
});

//Snapshot Testing
it("Matches snapshot", () => {
    const tree = renderer.create(<Button label={"Save"}></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});