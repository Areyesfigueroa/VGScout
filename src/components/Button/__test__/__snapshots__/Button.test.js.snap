// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Save
</div>
`;
